bundles/org.openhab.binding.groupepsa/src/main/java/org/openhab/binding/groupepsa/internal/rest/api/GroupePSAConnectApi.java

```diff
@@ -0,0 +1,200 @@
+    private final String realm;
+
+    protected final Gson gson;
+
+    public GroupePSAConnectApi(HttpClient httpClient, GroupePSABridgeHandler bridge, String clientId, String realm) {
+        this.httpClient = httpClient;
+        this.bridge = bridge;
+        this.clientId = clientId;
+        this.realm = realm;
+
+        gson = new GsonBuilder().registerTypeAdapterFactory(new GeometryAdapterFactory())
+                .registerTypeAdapter(ZonedDateTime.class, new JsonDeserializer<ZonedDateTime>() {
+                    @Override
+                    public @Nullable ZonedDateTime deserialize(JsonElement json, Type typeOfT,
+                            JsonDeserializationContext context) throws JsonParseException {
+                        return ZonedDateTime.parse(json.getAsJsonPrimitive().getAsString());
+                    }
+                }).registerTypeAdapter(Duration.class, new JsonDeserializer<Duration>() {
+                    @Override
+                    public @Nullable Duration deserialize(JsonElement json, Type typeOfT,
+                            JsonDeserializationContext context) throws JsonParseException {
+                        return Duration.parse(json.getAsJsonPrimitive().getAsString());
+                    }
+                }).create();
+    }
+
+    protected HttpClient getHttpClient() {
+        return httpClient;
+    }
+
+    protected GroupePSABridgeHandler getBridge() {
+        return bridge;
+    }
+
+    public String getBaseUrl() {
+        return API_URL;
+    }
+
+    private ContentResponse executeRequest(final String uri) throws GroupePSACommunicationException {
+        return executeRequest(uri, "application/hal+json");
+    }
+
+    static Throwable getRootCause(Throwable e) {
+        Throwable nextE;
+        do {
+            nextE = e.getCause();
+            if (nextE != null) {
+                e = nextE;
+            }
+        } while (nextE != null);
+        return e;
+    }
+
+    public ContentResponse executeRequest(final String uri, final String accept)
+            throws GroupePSACommunicationException {
+        Request request = getHttpClient().newRequest(uri);
+
+        String token = getBridge().authenticate();
+
+        request.timeout(10, TimeUnit.SECONDS);
+
+        request.param("client_id", this.clientId);
+
+        request.header("Authorization", "Bearer " + token);
+        request.header("Accept", accept);
+        request.header("x-introspect-realm", this.realm);
+
+        request.method(HttpMethod.GET);
+
+        logger.trace("HttpRequest {}", request.getURI());
+        logger.trace("HttpRequest Headers:\n{}", request.getHeaders());
+
+        try {
+            ContentResponse response = request.send();
+            logger.trace("HttpResponse {}", response);
+            logger.trace("HttpResponse Headers:\n{}", response.getHeaders());
+            logger.trace("HttpResponse Content: {}", response.getContentAsString());
```

Comment:
Can this message be replaced by using the debugger? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
Prefer to keep it in as it helps debugging other peoples setups.